homework/password-check/validation.cpp

```diff
@@ -1,2 +1,113 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+#include <iostream>
+#include <cctype>
+#include <map>
+
+std::string getErrorMessage(ErrorCode error_code){
+    std::string result;
+    
+    switch (error_code){
+    case ErrorCode::Ok :
+        result = "Ok";
+        break;
+    case ErrorCode::PasswordNeedsAtLeastNineCharacters :
+        result = "Password needs to have at least nine characters";
+        break;
+    case ErrorCode::PasswordNeedsAtLeastOneNumber : 
+        result = "Password needs to have at least one number";
+        break;
+    case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter :
+        result = "Password needs to have at least one special character";
+        break;
+    case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter :
+        result = "Password needs to have at least one uppercase letter";
+        break;
+    case ErrorCode::PasswordsDoNotMatch :
+        result = "Passwords do not match";
+        break;
+    default :
+    result = "Unknown error";
+    }
+
+    return result;
+
+}
+bool doPasswordsMatch(std::string password,std::string repeatedPassword){
+
+    if (password == repeatedPassword){
+        return true;
+    }else{
+        return false;
+    }
```

Comment:
Add space after `if`, `for`, `while` `function` etc...

Add space between operator like `+` `-` `/` `:` `&&` etc..